core/spot_instance_request.go

```diff
@@ -0,0 +1,78 @@
+package autospotting
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+type spotInstanceRequest struct {
+	*ec2.SpotInstanceRequest
+	region *region
+	asg    *autoScalingGroup
+}
+
+// This function returns an Instance ID
+func (s *spotInstanceRequest) waitForAndTagSpotInstance() {
+
+	logger.Println(s.asg.name, "Waiting for spot instance for spot instance request",
+		*s.SpotInstanceRequestId)
+
+	ec2Client := s.region.services.ec2
+
+	params := ec2.DescribeSpotInstanceRequestsInput{
+		SpotInstanceRequestIds: []*string{s.SpotInstanceRequestId},
+	}
+
+	err := ec2Client.WaitUntilSpotInstanceRequestFulfilled(&params)
+	if err != nil {
+		logger.Println(s.asg.name, "Error waiting for instance:", err.Error())
+		return
+	}
+
+	logger.Println(s.asg.name, "Done waiting for an instance.")
+
+	// Now we try to get the InstanceID of the instance we got
+	requestDetails, err := ec2Client.DescribeSpotInstanceRequests(&params)
+	if err != nil {
+		logger.Println(s.asg.name, "Failed to describe spot instance requests")
+	}
+
+	// due to the waiter we can now safely assume all this data is available
```

Comment:
Because people. :)
I was just pointing, that we can't be sure that all data is "safely" available, since someone or another program could have make a call in between.
Considering the unlikelihood of this happening, I honestly don't think it's worth adding a check, I make the same assumption in the 'configurable algorithm': where I don't double check the tags between 2 methods.